### PR TITLE
Add addon via yast2 and proxy SCC

### PR DESCRIPTION
**Pull request by dzedro**
[Add SDK+HA](http://10.100.12.155/tests/4959)
[Add SDK](http://10.100.12.155/tests/4962)

### CODE REVIEW

File: lib/registration.pm [hunk 1]
```diff
@@ -185,7 +185,18 @@ sub fill_in_registration_data {
...
                     next;
                 }
-                elsif (match_has_tag('inst-addon') || match_has_tag('yast_scc-pkgtoinstall')) {
+                elsif (match_has_tag('yast_scc-pkgtoinstall')) {
+                    send_key 'alt-a';
+                    assert_screen 'automatic-changes';
+                    send_key 'alt-o';
+                    if (check_screen 'unsupported-packages', 5) {
+                        send_key 'alt-o';
+                    }
```
Comment (okurz) [inline lib/registration.pm]: you know I don't like timeouts on check_screen which are not 0, so what about

```perl
wait_screen_change { send_key 'alt-o' };
send_key 'alt-o' if check_screen('unsupported-packages', 0);
```
?

File: tests/installation/addon_products_via_SCC_yast2.pm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        for my $addon (split(/,/, get_var('SCC_ADDONS', ''))) {
+            my $uc_addon = uc $addon;    # change to uppercase to match variable
+            push(@addon_proxy, "\b.$addon-" . get_var("BUILD_$uc_addon"));
+        }
+        script_sudo "echo \"@addon_proxy.proxy.scc.suse.de\" > /etc/SUSEConnect", 0;
+    }
+    type_string "killall xterm\n";
+    x11_start_program("xdg-su -c '/sbin/yast2'");
+    if ($password) { type_password; send_key "ret", 1; }
+    if (check_screen 'packagekit-warning', 5) {
```
Comment (okurz) [inline tests/installation/addon_products_via_SCC_yast2.pm]: see above

File: tests/installation/addon_products_via_SCC_yast2.pm [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        my @addon_proxy = ("url: http://server-" . get_var('BUILD_SLE'));
+        for my $addon (split(/,/, get_var('SCC_ADDONS', ''))) {
+            my $uc_addon = uc $addon;    # change to uppercase to match variable
+            push(@addon_proxy, "\b.$addon-" . get_var("BUILD_$uc_addon"));
+        }
+        script_sudo "echo \"@addon_proxy.proxy.scc.suse.de\" > /etc/SUSEConnect", 0;
+    }
+    type_string "killall xterm\n";
+    x11_start_program("xdg-su -c '/sbin/yast2'");
+    if ($password) { type_password; send_key "ret", 1; }
```
Comment (okurz) [inline tests/installation/addon_products_via_SCC_yast2.pm]: don't use `wait` option on `send_key` as `wait_idle` is deprecated.

File: tests/installation/addon_products_via_SCC_yast2.pm [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    if (get_var('SCC_ADDONS')) {
+        my @addon_proxy = ("url: http://server-" . get_var('BUILD_SLE'));
+        for my $addon (split(/,/, get_var('SCC_ADDONS', ''))) {
+            my $uc_addon = uc $addon;    # change to uppercase to match variable
+            push(@addon_proxy, "\b.$addon-" . get_var("BUILD_$uc_addon"));
+        }
+        script_sudo "echo \"@addon_proxy.proxy.scc.suse.de\" > /etc/SUSEConnect", 0;
+    }
+    type_string "killall xterm\n";
+    x11_start_program("xdg-su -c '/sbin/yast2'");
```
Comment (okurz) [inline tests/installation/addon_products_via_SCC_yast2.pm]: @Soulofdestiny invested some effort to use the same style "yast2" and not "/sbin/yast2" in a76bc39, can you change this?

File: tests/installation/addon_products_via_SCC_yast2.pm [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+# Summary: add addon to SLES via SCC
+# Maintainer: Jozef Pupava <jpupava@suse.com>
+
+use base "y2logsstep";
+use strict;
+use testapi;
+use registration 'fill_in_registration_data';
+
+sub run() {
+    x11_start_program('xterm');
```
Comment (okurz) [inline tests/installation/addon_products_via_SCC_yast2.pm]: I am not so sure about the module being in `tests/installation` as this one is not called as part of installation. you don't call this module anywhere anyway

Comment (dzedro) [inline tests/installation/addon_products_via_SCC_yast2.pm]: It is in installation same as e.g. [sles+rt](https://openqa.suse.de/tests/715106)

File: tests/installation/addon_products_via_SCC_yast2.pm (hunk 4 above)
Comment (dzedro) [inline tests/installation/addon_products_via_SCC_yast2.pm]: Running yast2 in root console is not same as via xdg-su, this has nothing with needles and it doesn't work without full path here

**Comment by dzedro** [comment]
correct tests:
[sles+sdk+ha+proxy_SCC](http://10.100.12.155/tests/4969)
[sles+sdk+proxy_SCC](http://10.100.12.155/tests/4968)

Updated

There are missing some stuff and I guess there is some duplicate code,
I don't want to touch it now, all this addons and registration need to [refactor](https://progress.opensuse.org/issues/16588)

**Comment by mitiao** [comment]
It broke registration with scc addons for om testings:
https://openqa.suse.de/tests/768686
https://openqa.suse.de/tests/768655
https://openqa.suse.de/tests/768657

Failure was caused by twice yast_scc-pkgtoinstall asserting after this applied.

Addons registration and installation in both of console and x11 should work by previous lib/registration.pm with calling yast_scc_registration from tty or xterm.

**Comment by dzedro** [comment]
@mitiao I see, fixing

**Comment by dzedro** [comment]
@mitiao created https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/2431

**Pull request merged**